module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -276,14 +317,49 @@ public function generate()
         }
 
         // Set-up Sitemap Index
-        $this->buildIndex($currentPage - 1);
+        $this->buildIndex($currentPage - 1, $additionalSitemaps);
 
         // Display total elapsed time in verbose mode:
         $this->verboseMsg(
             'Elapsed time (in seconds): ' . round($this->getTime() - $startTime)
         );
     }
 
+    /**
+     * Generate sitemaps with any enabled plugins
+     *
+     * @return array
+     */
+    protected function generateWithPlugins(): array
+    {
+        $additionalSitemaps = [];
+        if ($plugins = $this->config->Sitemap->plugins) {
+            $pluginSitemaps = [];
+            foreach ($plugins->toArray() as $pluginName) {
+                $plugin = $this->getPlugin($pluginName);
+                $sitemapName = $plugin->getSitemapName();
+                $this->verboseMsg(
+                    "Generating sitemap '$sitemapName' with '$pluginName'"
+                );
+                if (!isset($pluginSitemaps[$sitemapName])) {
+                    $pluginSitemaps[$sitemapName] = $this->getNewSitemap();
+                }
+                $plugin->addUrls($pluginSitemaps[$sitemapName]);
+            }
+
+            foreach ($pluginSitemaps as $sitemapName => $sitemap) {
+                if (!$sitemap->isEmpty()) {
+                    $filename = $this->getFilenameForPage($sitemapName);
```

Comment:
This may be fine for now, but do we potentially need to worry about plugins that generate huge numbers of URLs? Should we have a mechanism for splitting a huge list into multiple files?

Reply:
Good point. Now added and refactored plugins to use yield.